src/serialize/root.zig

```diff
@@ -0,0 +1,133 @@
+const std = @import("std");
+const testing = std.testing;
+const Allocator = std.testing.allocator;
+
+const MAX_FIELD_COUNT: u8 = 255;
+
+pub fn tomlize(allocator: std.mem.Allocator, obj: anytype, writer: anytype) !void {
+    try serializeStruct(allocator, obj, writer);
+}
+
+fn serializeStruct(allocator: std.mem.Allocator, value: anytype, writer: anytype) !void {
+    const ttype = @TypeOf(value);
+    const tinfo = @typeInfo(ttype);
+    if (!std.mem.eql(u8, @tagName(tinfo), "struct")) @panic("non struct type given to serialize");
```

Comment:
Consider something like
```zig
    switch (tinfo) {
        .@"struct" => {},
        else => return error.InvalidTopType,
    }
```

Reply:
I am not sure about it. The method is not marked public and hence it won't be called by the user directly. So the only way this function can have errors is if us devs make a error by sending the wrong type to it. In such case it should be a hard panic and not something that the end users are going to deal with.

Reply:
Makes sense, agreed 👍 Let's just replace the type name comparison with the enum comparison.